source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -108,46 +105,63 @@ void AuthenticatorImpl::startVerify() {
   while (!tokens_.empty()) {
     curr_token_ = std::move(tokens_.back());
     tokens_.pop_back();
-    jwt_ = {};
-    status = jwt_.parseFromString(curr_token_->token());
+    jwt_ = std::make_unique<::google::jwt_verify::Jwt>();
```

Comment:
jwt_.reset();

Reply:
there is no reset method.

Reply:
Isn't reset() the same thing as setting the pointer to nullptr? wouldnt that segfault?

Reply:
getting segfault when it's jwt_.reset(). Is jwt_.reset(new ::google::jwt_verify::Jwt()); better than the assignment operator?

Reply:
Sorry,  my bad.  I mis-read the code.  I through you were just clear the unique_ptr.  Actually you are creating a new one.   So your code is fine.